collections/folders.go

```diff
@@ -0,0 +1,105 @@
+package collections
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+	"github.com/textileio/go-threads/api/client"
+	s "github.com/textileio/go-threads/store"
+)
+
+type Folder struct {
+	ID        string
+	Path      string
+	Name      string
+	Public    bool
+	ProjectID string
+	Created   int64
+}
+
+type Folders struct {
+	threads *client.Client
+	storeID *uuid.UUID
+	token   string
+}
+
+func (f *Folders) GetName() string {
+	return "Folder"
+}
+
+func (f *Folders) GetInstance() interface{} {
+	return &Folder{}
+}
+
+func (f *Folders) GetIndexes() []*s.IndexConfig {
+	return []*s.IndexConfig{{
+		Path: "Path",
+	}, {
+		Path:   "Name",
+		Unique: true,
```

Comment:
A project can only have one folder by a given name. Question: Maybe we can use "bucket" instead of "folder"? Too close to S3?